src/Base/Exception.h

```diff
@@ -41,23 +48,45 @@ class BaseExport Exception : public BaseClass
   TYPESYSTEM_HEADER();
 
 public:
+
   virtual ~Exception() throw() {}
 
   Exception &operator=(const Exception &inst);
+
   virtual const char* what(void) const throw();
-  virtual void ReportException (void) const;
+
+  void ReportException (void) const;

```

Comment:
I dislike it to make ReportException() non-virtual and add a new virtual method report() because it's a regression in flexibility.
In FileException report() is overwritten but ReportException() still writes a message of the form:
"Exception (<time>): <message>
Better would be to overwrite ReportException() in FileException and one could explicitly write "File exception" or "IO exception" or the like.
The class PyException is the link between Base::Exception and Python exceptions and there it's needed to implement ReportException differently.